hermes-frontend/src/test/java/pl/allegro/tech/hermes/frontend/publishing/MessageContentTypeEnforcerTest.java

```diff
@@ -2,38 +2,30 @@
 
 import org.apache.avro.Schema;
 import org.junit.Test;
-import pl.allegro.tech.hermes.api.ContentType;
-import pl.allegro.tech.hermes.api.Topic;
-import pl.allegro.tech.hermes.domain.topic.schema.SchemaRepository;
+import pl.allegro.tech.hermes.domain.topic.schema.SchemaVersion;
+import pl.allegro.tech.hermes.domain.topic.schema.CompiledSchema;
+import pl.allegro.tech.hermes.frontend.publishing.avro.AvroMessage;
 import pl.allegro.tech.hermes.frontend.publishing.message.Message;
 import pl.allegro.tech.hermes.test.helper.avro.AvroUser;
 
 import java.io.IOException;
 
 import static org.assertj.core.api.Assertions.assertThat;
-import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.when;
-import static pl.allegro.tech.hermes.test.helper.builder.TopicBuilder.topic;
 
 public class MessageContentTypeEnforcerTest {
 
-    @SuppressWarnings("unchecked")
-    private SchemaRepository<Schema> schemaRepository = mock(SchemaRepository.class);
-
-    private MessageContentTypeEnforcer enforcer = new MessageContentTypeEnforcer(schemaRepository);
+    private MessageContentTypeEnforcer enforcer = new MessageContentTypeEnforcer();
 
     private AvroUser avroMessage = new AvroUser("Bob", 30, "black");
+    private CompiledSchema<Schema> schema = new CompiledSchema<>(avroMessage.getSchema(), SchemaVersion.valueOf(0));
 
     @Test
     public void shouldConvertToAvroWhenReceivedJSONOnAvroTopic() throws IOException {
         // given
-        Topic topic = topic("enforcer", "json2avro").withContentType(ContentType.AVRO).build();
-        Message message = new Message("1", avroMessage.asJson().getBytes(), 1234);
-
-        when(schemaRepository.getSchema(topic)).thenReturn(avroMessage.getSchema());
+        AvroMessage message = new AvroMessage("1", avroMessage.asJson().getBytes(), 1234, schema);
```

Comment:
The idea behind this is that if AvroMessage contains json bytes then it means that it is invalid and we need to enforce avro bytes. Reasoning behind using AvroMessage as a container for json bytes is that AvroMessage has a place for avro Schema which can be used to express what we expect for this message to conform. Look here to see the entire flow: ```MessageFactory.createAvroMessage```.